bitbots_body_behavior/src/bitbots_body_behavior/decisions/dribble_or_kick.py

```diff
@@ -0,0 +1,54 @@
+import rospy
+
+from dynamic_stack_decider.abstract_decision_element import AbstractDecisionElement
+
+
+class DribbleOrKick(AbstractDecisionElement):
+    def __init__(self, blackboard, dsd, parameters=None):
+        super().__init__(blackboard, dsd, parameters)
+        self.orient_threshold = self.blackboard.config['dribble_orient_threshold']
+        self.goal_distance_threshold = self.blackboard.config['dribble_goal_distance_threshold']
+        self.ball_distance_threshold = self.blackboard.config['dribble_ball_distance_threshold']
+
+        self.kick_length = rospy.get_param('behavior/body/kick_cost_kick_length')
+        self.angular_range = rospy.get_param('behavior/body/kick_cost_angular_range')
+        self.max_kick_angle = rospy.get_param('behavior/body/max_kick_angle')
+
+    def perform(self, reevaluate=False):
+        """
+        Determines whether we want to dribble if the area in front of us is clear, or to kick
+        :param reevaluate:
+        :return:
+        """
+        # robot needs to be correctly aligned to ball, so that opponent goal is in front
+        goal_angle = abs(self.blackboard.world_model.get_map_based_opp_goal_angle())
```

Comment:
This checks only the angle of the vector between the goal and the ball. Not the orientation of our robot relative to that. This would result in our robot dribbling in the wrong direction if we e.g. approach the ball from the other side.